Classes/Event/Listener/AfterLinkIsGeneratedListener.php

```diff
@@ -70,4 +80,92 @@ public function __invoke(AfterLinkIsGeneratedEvent $event): void
             $event->setLinkResult($result);
         }
     }
+
+    private function getTargetSite(AfterLinkIsGeneratedEvent $event): Site
+    {
+        $linkConfiguration = $event->getLinkResult()->getLinkConfiguration();
+
+        if (isset($linkConfiguration['parameter.'])) {
+            // Evaluate "parameter." stdWrap but keep additional information (like target, class and title)
+            $linkParameterParts = $this->typoLinkCodecService->decode($linkConfiguration['parameter'] ?? '');
+            $modifiedLinkParameterString = $event->getContentObjectRenderer()->stdWrap($linkParameterParts['url'], $linkConfiguration['parameter.']);
+            // As the stdWrap result might contain target etc. as well again (".field = header_link")
+            // the result is then taken from the stdWrap and overridden if the value is not empty.
+            $modifiedLinkParameterParts = $this->typoLinkCodecService->decode((string)($modifiedLinkParameterString ?? ''));
+            $linkParameterParts = array_replace($linkParameterParts, array_filter($modifiedLinkParameterParts, static fn($value) => trim((string)$value) !== ''));
+            $linkParameter = $this->typoLinkCodecService->encode($linkParameterParts);
+        } else {
+            $linkParameter = trim((string)($linkConfiguration['parameter'] ?? ''));
+        }
+
+        try {
+            [$linkParameter] = $this->resolveTypolinkParameterString($linkParameter, $linkConfiguration);
+        } catch (UnableToLinkException $e) {
+            $this->logger->warning($e->getMessage(), ['linkConfiguration' => $linkConfiguration]);
+            throw $e;
+        }
+        $linkDetails = $this->resolveLinkDetails($linkParameter, $linkConfiguration, $event->getContentObjectRenderer());
+        if ($linkDetails === null) {
+            throw new UnableToLinkException('Could not resolve link details from ' . $linkParameter, 1642001442, null, $event->getLinkResult()->getLinkText());
+        }
+
+        if (($linkDetails['pageuid'] ?? '') === 'current') {
+            return $event->getContentObjectRenderer()->getRequest()->getAttribute('site');
+        }
+
+        return $this->siteFinder->getSiteByPageId((int)$linkDetails['pageuid']);
```

Comment:
@twoldanski, for custom typolinks `pageuid` isn't set and triggers a `PHP Warning: Undefined array key "pageuid"`. Maybe use `($linkDetails['pageuid'] ?? 'current') === 'current')` in line 112 to prevent that?

Reply:
See separate issue: https://github.com/TYPO3-Headless/headless/issues/762